File: pyg-sellergoods-interface/src/main/java/com/pyg/sellergoods/service/GoodsDescService.java
package com.pyg.sellergoods.service;
import java.util.List;
import com.pyg.pojo.TbGoodsDesc;

import PageBean.PageResult;
/**
 * 服务层接口
 * @author Administrator
 *
 */
public interface GoodsDescService {

	/**
	 * 返回全部列表
	 * @return
	 */
	public List<TbGoodsDesc> findAll();
	

	
	
	/**
	 * 增加
	*/
	public boolean add(TbGoodsDesc goodsDesc);
	
	
	/**
	 * 修改
	 */
	public boolean update(TbGoodsDesc goodsDesc);
	

	/**
	 * 根据ID获取实体
	 * @param id
	 * @return
	 */
	public TbGoodsDesc findOne(Long id);
	
	
	/**
	 * 批量删除
	 * @param ids
	 */
	public void delete(Long[] ids);

	/**
	 * 分页
	 * @param page 当前页 码
	 * @param rows 每页记录数
	 * @return
	 */
	PageResult search(TbGoodsDesc goodsDesc, int page, int rows);
}

File: pyg-manager-web/src/main/webapp/js/Service/goodsDescService.js
//服务层
app.service('goodsDescService',function($http){

    //读取列表数据绑定到表单中
    this.findAll = function () {
        return $http.get('../goodsDesc/findAll');
    }
    //存储&修改
    this.save = function (entity) {
        return $http.post('../goodsDesc/save', entity);
    }

    //查询实体
    this.findOne = function (id) {
        return $http.get('../goodsDesc/findOne?id=' + id);
    }

    //删除
    this.dele = function (ids) {
        return $http.get('../goodsDesc/delete?ids=' + ids);
    }


    //分页 查询
    this.findPage = function (page, size, searchEntity) {
        return $http.post('../goodsDesc/search?page=' + page + '&size=' + size, searchEntity);
    }
});
